test/extended/router/h2spec.go

```diff
@@ -0,0 +1,289 @@
+package router
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	routeclientset "github.com/openshift/client-go/route/clientset/versioned"
+	"github.com/openshift/origin/test/extended/router/h2spec"
+	exutil "github.com/openshift/origin/test/extended/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+type h2specFailingTest struct {
+	TestCase   *h2spec.JUnitTestCase
+	TestNumber int
+}
+
+type h2specRouteTypeTest struct {
+	routeType     string
+	hostname      string
+	knownFailures map[string]bool
+}
+
+func (f h2specFailingTest) ID() string {
+	return fmt.Sprintf("%s.%d", f.TestCase.Package, f.TestNumber)
+}
+
+func (t h2specRouteTypeTest) h2specCommand(results string) string {
+	return fmt.Sprintf("h2spec --tls --insecure --strict --host=%s --junit-report=%s", t.hostname, results)
+}
+
+var _ = g.Describe("[sig-network-edge][Conformance][Area:Networking][Feature:Router]", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		configPath = exutil.FixturePath("testdata", "router", "router-h2spec.yaml")
+		oc         = exutil.NewCLI("router-h2spec", exutil.KubeConfigPath())
+	)
+
+	// this hook must be registered before the framework namespace teardown
+	// hook
+	g.AfterEach(func() {
+		if g.CurrentGinkgoTestDescription().Failed {
+			client := routeclientset.NewForConfigOrDie(oc.AdminConfig()).RouteV1().Routes(oc.KubeFramework().Namespace.Name)
+			if routes, _ := client.List(metav1.ListOptions{}); routes != nil {
+				outputIngress(routes.Items...)
+			}
+			exutil.DumpPodLogsStartingWith("h2spec", oc)
+		}
+	})
+
+	g.By(fmt.Sprintf("creating routes from a config file %q", configPath))
+
+	g.Describe("The HAProxy router", func() {
+		g.It("should pass the h2spec conformance tests", func() {
+			var err error
+
+			_, err = exutil.WaitForRouterServiceIP(oc)
```

Comment:
Not sure if this check matters since you're waiting for route status (implying router processed the route) and since the tests won't have started without a ready cluster anyway

Reply:
Ack. Will drop this.